File: Backend/ProjectHub/src/main/java/com/projecthub/service/TasksService.java
package com.projecthub.service;

import java.util.List;
import java.util.Map;

import com.projecthub.model.Tasks;

public interface TasksService {

	public Tasks saveTasks(Tasks task);
	public Tasks getTaskById(Long id);
	public Map<String, Object> getAllTasks();
	public Tasks updateTasks(Long id , Tasks task);
	public Map<String, Object> deleteTaskById(Long id);
	
}

File: Ang-Fronted/Project-Grid/src/app/services/tasks.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { CreateTask } from '../modals/task';
import { Observable } from 'rxjs';
import { Utility } from './utilis.service';

@Injectable({
  providedIn: 'root'
})
export class TasksService {

  baseTaskUrl =  Utility.baseUrl+"/api/tasks"

  constructor(
    private http : HttpClient
  ) { }

  createTask(task:CreateTask,token:string):Observable<any>{
    const headers = new HttpHeaders({
      'Authorization': `Bearer ${token}` 
    });
      return this.http.post<any>(
        this.baseTaskUrl+'/createTask',
        task ,
        {headers}
      );
  }
  getAllTasks(token:string):Observable<any>{
    const headers = new HttpHeaders({
      'Authorization': `Bearer ${token}`
    });
    return this.http.get<any>(this.baseTaskUrl+"/getAllTasks",{headers});
  }
  deleteTaskById(id:number,token:string):Observable<any>{
    const headers = new HttpHeaders({
      'Authorization': `Bearer ${token}`
    });
    return this.http.delete<any>(this.baseTaskUrl+`/deleteTask/${id}`,{headers},);
  }

}
